examples/helloworld/priv/client.exs

```diff
@@ -1,3 +1,10 @@
+defmodule Foo do
+  def call(service_mod, rpc, path, channel, request, opts, next) do
+    IO.puts("Foo")
+    next.(service_mod, rpc, path, channel, request, opts)
+  end
+end
+
 {:ok, channel} = GRPC.Stub.connect("localhost:50051")
-{:ok, reply} = channel |> Helloworld.Greeter.Stub.say_hello(Helloworld.HelloRequest.new(name: "grpc-elixir"))
+{:ok, reply} = channel |> Helloworld.Greeter.Stub.say_hello(Helloworld.HelloRequest.new(name: "grpc-elixir"), middlewares: [Foo])
```

Comment:
The usage of client middleware is a little inconvenient. I wouldn't like to pass middleware to every function call😂

Reply:
Yeah it's not very convenient indeed 😅. 

This can be improved but would require to add a `middleware: [A,B,C]` option to the `Stub` callback module , which happens to be in generated files (i.e. `helloworld.pb.ex`) so I didn't go straight for it. 

Do you think that's an acceptable tradeoff?

Reply:
Middlewares should designs like the Phoenix `pipe`.
Requests go through all the middlewares in a pipeline.